scripts/check_e2e_tests_are_captured_in_ci.py

```diff
@@ -0,0 +1,145 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A script to check that travis.yml file & protractor.conf.js have the
+same test suites."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+import re
+
+import python_utils
+import utils
+
+
+def extract_travis_jobs_section():
+    """Extracts the env/jobs section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of all jobs in travis.yml file.
+    """
+    travis_file = read_travis_yml_file()
+    jobs_raw = travis_file['env']['jobs']
+    suites_from_jobs = []
+    for job in jobs_raw:
+        # The jobs are in the form RUN_E2E_TESTS_ACCESSIBILITY=true
+        # so the following line removes the RUN_E2E_TESTS_(14 length)
+        # and =true(5 length) part.
+        suites_from_jobs.append(utils.snake_case_to_camel_case(job[14:-5].lower()))
+    suites_from_jobs.sort()
+
+    return suites_from_jobs
+
+
+def extract_travis_script_section():
+    """Extracts the script section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of all scripts in travis.yml file.
+    """
+    travis_file = read_travis_yml_file()
+    script_raw = travis_file['script']
+    # The following line extracts the test suites from patterns like
+    # --suite="accessibility" --
+    hyphen_between_regex = re.compile(r'--suite="(.+?)" --')
+    suites_from_script = []
+
+    for script in script_raw:
+        matches = hyphen_between_regex.finditer(script)
+        for match in matches:
+            # The extracted data is of the form --suite="subscriptions" --
+            # hence 9 to remove --suite=" and -4 to remove " --
+            suites_from_script.append(match.group()[9:-4])
+    suites_from_script.sort()
+
+    return suites_from_script
+
+
+def extract_protractor_test_suites():
+    """Extracts the test suites section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of all test suites in protractor.conf.js file.
+    """
+    protractor_config_file = read_protractor_conf_file()
+    # The following line extracts suite object from protractor.conf.js.
+    suite = re.compile(r'suites = {([^}]+)}').findall(protractor_config_file)[0]
+
+    # The following line extracts the keys/test suites from the suites object.
+    key_regex = re.compile(r'\s(.*?):')
+    protractor_suites = []
+    for match in key_regex.finditer(suite):
+        # Since the keys are in the form adminPage: so -1 to remove the
+        # colon part :
+        protractor_suites.append(match.group()[:-1].strip())
+
+    protractor_suites.sort()
+    return protractor_suites
+
+
+def read_protractor_conf_file():
+    """Returns the core/tests/protractor.conf.js. file.
+
+    Returns:
+        str. The protractor.conf.js as a string.
```

Comment:
The contents of protractor.conf.js, as a string.

Reply:
Done